sdk/helper/tlsutil/tlsutil.go

```diff
@@ -110,3 +115,72 @@ func ClientTLSConfig(caCert []byte, clientCert []byte, clientKey []byte) (*tls.C
 
 	return tlsConfig, nil
 }
+
+func SetupTLSConfig(conf map[string]string, address string) (*tls.Config, error) {
+	serverName, _, err := net.SplitHostPort(address)
+	switch {
+	case err == nil:
+	case strings.Contains(err.Error(), "missing port"):
+		serverName = conf["address"]
+	default:
+		return nil, err
+	}
+
+	insecureSkipVerify := false
+	tlsSkipVerify, ok := conf["tls_skip_verify"]
```

Comment:
nit: can omit `ok` here since `tlsSkipVerify` will be `""` if not present (and the if block skipped).

Reply:
Oh sure. That code was copy pasta so I didn't scrutinize it too much.

Reply:
Implemented in https://github.com/hashicorp/vault/pull/8123/commits/2b0f648391e270e6c60ecfdbcd67f76c75170133